src/main/scala/org/allenai/pnp/semparse/EntityLinking.scala

```diff
@@ -75,4 +77,13 @@ case class Span(val start: Int, val end: Int) {
   def contains(index: Int): Boolean = {
     index >= start && index < end
   }
+}
+
+case class KnowledgeGraph(val nodePairs: List[(String, String)]) {
+  val nodeSet = nodePairs.flatMap(p => List(p._1, p._2)).toSet
```

Comment:
I think you just want `nodePairs.flatten.toSet` here.  Except it's already flat.  I'm really not sure what this `flatMap` is doing - looks like it's a no-op, and `nodePairs.toSet` would work.